types/yup/index.d.ts

```diff
@@ -153,6 +156,8 @@ export interface NumberSchema<T extends number | null = number>
     round(type: "floor" | "ceil" | "trunc" | "round"): NumberSchema<T>;
     nullable(isNullable?: true): NumberSchema<T | null>;
     nullable(isNullable: false): NumberSchema<Exclude<T, null>>;
```

Comment:
(reposting comment on PR instead of commit)
can nullable be called with anything but a literal? It doesn't work with an unknown boolean:

```ts
function wrapper(b: boolean, msx: MixedSchema) {
  return msx.nullable(b)
}
```

Gives an error "Argument of type 'boolean' is not assignable to 'false'."

Reply:
We might need to flip the order of these function declarations for the types to flow correctly?

Reply:
I think you actually need another overload for `boolean`, but I'm not sure what to return. Just `NumberSchema<T>`?

Reply:
@sandersn 	

Adding an extra overload to the `MixedSchema`:
```TypeScript
export interface MixedSchema<T = any> extends Schema<T> {
    nullable(isNullable?: boolean): MixedSchema<T>;
    nullable(isNullable?: true): MixedSchema<T | null>;
    nullable(isNullable: false): MixedSchema<Exclude<T, null>>;
    required(message?: TestOptionsMessage): MixedSchema<Exclude<T, undefined>>;
    notRequired(): MixedSchema<T | undefined>;
}
```

And overloading the wrapper like:
```TypeScript
function wrapper<T>(b: false, msx: MixedSchema<T>): MixedSchema<T>;
function wrapper<T>(b: true, msx: MixedSchema<T>): MixedSchema<T | null>;

function wrapper<T>(b: boolean, msx: MixedSchema<T>) {
    return msx.nullable(b);
}
``` 

Should IMHO result in the correct typing:
```TypeScript
const resultingSchema1 = wrapper<string | number>(false, yup.mixed().oneOf(['1', 2]));
const resultingSchema2 = wrapper<string | number>(true, yup.mixed().oneOf(['1', 2]));
```

Resulting type signatures:
```TypeScript
const resultingSchema1: yup.MixedSchema<string | number>
const resultingSchema2: yup.MixedSchema<string | number | null>
```

Reply:
Shouldn't the boolean overload go last since it will otherwise match literal `true` and `false`?

Reply:
Yeah, I'm thinking this too.

Reply:
Agreed, added the various overloads and a test for them.